### PR TITLE
PUBDEV-3274 - Updates to GBM section in User Guide

### PR DESCRIPTION
**Pull request by angela0xdata**
Added “laplace” and “quantile” to list of distribution options. Removed
duplicated options in the bulleted list of distribution options.
Updated FAQ for the algorithm to use when the dataset has a large
number of columns.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/119)

### CODE REVIEW

File: h2o-docs/src/product/data-science/gbm.rst [hunk 1]
```diff
@@ -326,7 +316,7 @@ FAQ
 
 -  **What if there are a large number of columns?**
 
-  DRF models are best for datasets with fewer than a few thousand columns.
+  GBM models are best for datasets with a large number of columns (several thousand or more). DRF models are best for datasets with fewer than a few thousand columns.
```
Comment (arnocandel) [inline h2o-docs/src/product/data-science/gbm.rst]: not true - all tree methods are getting less useful with growing number of columns, DRF and GBM are the same in that respect

File: h2o-docs/src/product/data-science/gbm.rst [hunk 2]
```diff
@@ -102,20 +101,11 @@ Defining a GBM Model
...
        -  If the distribution is **gaussian**, the response column must
           be numeric.
-       -  If the distribution is **multinomial**, the response column
-          must be categorical.
-       -  If the distribution is **poisson**, the response column must
-          be numeric.
-       -  If the distribution is **gamma**, the response column must be
-          numeric.
-       -  If the distribution is **tweedie**, the response column must
-          be numeric.
```
Comment (arnocandel) [inline h2o-docs/src/product/data-science/gbm.rst]: gaussian, laplace, quantile, poisson, gamma, tweedie all require numeric response

File: h2o-docs/src/product/data-science/gbm.rst [hunk 3]
```diff
@@ -102,20 +101,11 @@ Defining a GBM Model
...
-       -  If the distribution is **gaussian**, the response column must
-          be numeric.
        -  If the distribution is **laplace**, the data must be numeric
           and continuous (**Int**).
        -  If the distribution is **quantile**, the data must be numeric
           and continuous (**Int**).
```
Comment (arnocandel) [inline h2o-docs/src/product/data-science/gbm.rst]: continuous and integer? doesn't make sense to me

**Comment by angela0xdata** [comment]
I updated the FAQ and description of the laplace and quantile options.

File: h2o-docs/src/product/data-science/gbm.rst [hunk 4]
```diff
@@ -102,20 +101,11 @@ Defining a GBM Model
...
-          must be categorical.
-       -  If the distribution is **poisson**, the response column must
-          be numeric.
-       -  If the distribution is **gamma**, the response column must be
-          numeric.
-       -  If the distribution is **tweedie**, the response column must
-          be numeric.
-       -  If the distribution is **gaussian**, the response column must
+       -  If the distribution is **laplace**, the response column must
+          be numeric..
```
Comment (arnocandel) [inline h2o-docs/src/product/data-science/gbm.rst]: numeric.. -> numeric.

**Comment by angela0xdata** [comment]
Fixed the typo.

**Pull request merged**